apps/src/applab/appStorage.js

```diff
@@ -244,6 +246,35 @@ var handleDeleteRecord = function(tableName, record, onComplete, onError) {
   onComplete(true);
 };
 
+var recordListener = new RecordListener();
+
+/**
+ * This is a partial implementation of onRecordEvent in the following ways:
+ * 1. it polls instead of properly listening for notifications when data changes;
+ * 2. it only issues callbacks when records are created (not updated or deleted);
+ * 3. it assumes that record ids are strictly increasing. This is not currently true but
+ *    would become so if we decide to implement
+ *    https://www.pivotaltracker.com/story/show/110169770
+ * @param tableName Table to listen to.
+ * @param onRecord Callback to call when a record is added to the table.
+ * @param onError Callback to call with an error to show to the user.
+ */
+AppStorage.onRecordEvent = function(tableName, onRecord, onError) {
+  if (!tableName) {
+    onError('Error listening for record events: missing required parameter "tableName"');
+    return;
+  }
+
```

Comment:
isOnError required?  If not we should document; if so we should throw an error if it isn't defined.